flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/LaunchableMesosWorker.java

```diff
@@ -259,9 +262,11 @@ public String toString() {
 				throw new IllegalStateException("unsupported container type");
 		}
 		if(containerInfo != null) {
+			containerInfo.addAllVolumes(params.containerVolumes());
```

Comment:
There's an unnecessary restriction here, that volumes may be used only if a container image is also used.   I bet you can use volumes with the Mesos containerizer without using an image.     The code in the `case MESOS:` block should be reorganized to always set `containerInfo = Protos.ContainerInfo.newBuilder()`.  This way, `containerInfo` will never be null.